clients/TypeScript/packages/client/src/ChainSync/ChainSyncClient.ts

```diff
@@ -0,0 +1,97 @@
+import { Block, Ogmios, Point, Tip } from '@cardano-ogmios/schema'
+import {
+  ConnectionConfig,
+  Mirror,
+  InteractionContext,
+  createClientContext
+} from '../Connection'
+import { UnknownResultError } from '../errors'
+import { baseRequest } from '../Request'
+import { createPointFromCurrentTip, ensureSocketIsOpen } from '../util'
+import { findIntersect, Intersection } from './findIntersect'
+
+export interface ChainSyncClient {
+  context: InteractionContext
+  findIntersect: (points: Point[]) => ReturnType<typeof findIntersect>
+  initialIntersection: Intersection
+  on: (messageHandlers: {
+    rollBackward: (response: {
+      point: Point,
+      tip: Tip,
+      reflection: Mirror
+    }) => void
+    rollForward: (response: {
+      block: Block,
+      tip: Tip,
+      reflection: Mirror
+    }) => void
+  }) => void
+  requestNext: (options?: { mirror?: Mirror }) => void
+  shutdown: () => Promise<void>
+}
+
+export const createChainSyncClient = async (options?: {
+  connection?: ConnectionConfig
+}): Promise<ChainSyncClient> => {
+  return new Promise((resolve, reject) => {
+    createClientContext(options).then(context => {
+      const { socket } = context
+      socket.once('error', reject)
+      socket.once('open', async () => {
+        const initialIntersection = await findIntersect(
+          [await createPointFromCurrentTip({ socket, closeOnCompletion: false })],
```

Comment:
I wish there was a special `tip` point. I've been considering implementing it just in Ogmios, but having this in the client is perhaps preferable.

Reply:
Is there a chance of having it implemented at the protocol level, so that Ogmios stays aligned?

Reply:
I don't see this happening :/ ...